max.go

```diff
@@ -0,0 +1,116 @@
+package funk
+
+import (
+	"reflect"
+	"strings"
+)
+
+// Max checks for the type of input, validates the input, compares the elements and returns the maximum element in an array/slice.
+// It accepts []int, []int8, []int16, []int32, []int64, []float32, []float64, []string
+// For strings, the first different character amongst the elements is compared using go's inbuilt strings.Compare method
+// It returns int or string or nil
+// It returns nil for the following cases:
+//  - input is not an array/slice
+//  - input has heterogeneous elements
+func Max(input interface{}) interface{} {
```

Comment:
Maybe we should use Max(input ...interface{}) instead? It would be great if something like this worked:
```go
three := funk.Max(1, 2, 3)
```

Reply:
If we use the variadic function, with `interface{}` as acceptable type, the caller won't be able to use spread operator unless the array is declared as `[]interface{}`.
Example: 
Calling the function declared as `func Max(input ...interface{}) interface{}` will fail at:
`arr1 := []int{3,4,5,6}`
`six := funk.Max(arr1...)`  <---- Breaks here because `arr1` is `[]int`, not `[]interface{}`
The caller will have to individually pass the args.

Is that an acceptable limitation? How do you suggest we handle that?

Reply:
Yeah, it is not cool :( 

Maybe we should keep it as it is

Reply:
Hey
Any updates on this?